pkg/controller/storagecluster/initialization_reconciler_test.go

```diff
@@ -178,6 +180,30 @@ func TestInitStorageClusterResourcesUpdate(t *testing.T) {
 	assertExpectedResources(t, reconciler, cr, request)
 }
 
+func TestEnsureExternalStorageClusterResources(t *testing.T) {
+	cr := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalClusterDetailsSecret,
+			Namespace: "openshift-storage",
+		},
+		Data: map[string][]byte{
+			"data": []byte(jsonBlob),
+		},
+	}
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      externalClusterDetailsSecret,
+			Namespace: "openshift-storage",
+		},
+	}
+
+	reconciler := createFakeInitializationStorageClusterReconciler(t, cr)
+	err := reconciler.client.Create(nil, cr)
+	result, err := reconciler.Reconcile(request)
+	assert.NoError(t, err)
+	assert.Equal(t, reconcile.Result{}, result)
+}
```

Comment:
Follow the pattern in `TestInitStorageClusterResourcesUpdate()`:

1. Do a `reconciler.client.Create()` call for every resource we expect to be created, ensure no errors
2. Run `reconciler.Reconcile()` as you do now
3. Create a new function called `assertExpectedExternalResources()`
4. Within that function, use the `var externalResources` I suggested you make, iterate through the list, and for each item:
    1. Create an empty struct of the appropriate type (ConfigMap, Secret, StorageClass)
    2. Run `reconciler.client.Get()` into the empty struct
    3. Assert that the appropriate data from the `ExternalResource` item made it into the resource you just fetched